File: cpp/pascals_triangle.cpp
class Solution {
public:
    vector<vector<int> > generate(int numRows) {
        vector< vector<int> > result;
        for (vector<vector<int>>::size_type row=0; row!=numRows; ++row)
        {
            vector<int> tmp;
            for (vector<int>::size_type index=0; index!=row+1; ++index)
            {
                if(index==0 || index==row)
                    tmp.push_back(1);
                else
                    tmp.push_back(result[row-1][index-1]+result[row-1][index]);
            }
            result.push_back(tmp);
        }
        return result;
    }
};

File: python/pascals_triangle.py
class Solution:
    # @return a list of lists of integers
    def generate(self, numRows):
        result = []
        for row in range(numRows):
            tmp = []
            for index in range(0, row+1):
                if index == 0 or index == row:
                    tmp.append(1)
                else:
                    tmp.append(result[row-1][index-1] + result[row-1][index])
            result.append(tmp)
        return result